### PR TITLE
[SW-158] Support for Spark DataSet

### PR DESCRIPTION
**Pull request by vpatryshev**
This stuff is working; I'll need to work on code deduplication, can be done later.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/H2OContext.scala [hunk 1]
```diff
@@ -61,6 +61,7 @@ class H2OContext private (@transient val sparkContext: SparkContext, @transient
   with Serializable with SparkDataFrameConverter with SupportedRDDConverter with H2OContextUtils{
   self =>
 
+  val sqlc: SQLContext = SQLContext.getOrCreate(sparkContext)
```
Comment (mmalohlava) [inline core/src/main/scala/org/apache/spark/h2o/H2OContext.scala]: I am changing this val into `@transient` to make sure we are not capturing any non-serializable entities.

Comment (vpatryshev) [inline core/src/main/scala/org/apache/spark/h2o/H2OContext.scala]: Oh. Thanks a lot. The class is serializable.

**Comment by vpatryshev** [comment]
Wow!

**Comment by mmalohlava** [comment]
Thx @vpatryshev! (i squashed that, test that on yarn, and merge)

**Pull request merged**